happi/client.py

```diff
@@ -482,6 +514,11 @@ def __iter__(self):
     def __len__(self):
         return len(self.all_items)
 
+    def _maybe_clear_cache(self):
+        """Clear the backend cache if not in a retain-cache block."""
+        if not self._retain_cache:
+            self.backend.clear_cache()
```

Comment:
Did we still want a timeout condition here?  As it is now I think the cache will never clear if we set the client to retain the cache?

Reply:
The timeout is no longer necessary as the basis for clearing the cache is "did the user ask to clear the cache?"

The control that the user can leverage is only by way of:
1. Calling `search()` et al on their own (cache clearing on every call)
2. Using the `retain_cache_context` (cache clearing at most once per context)